oss-cloud-backend/gitHubApiService.js

```diff
@@ -67,13 +70,84 @@ searchUserPullRequests = async(username, repo) => {
   do {
     await octokit.search.issuesAndPullRequests({
       q: "repo:" + repo.owner.login + "/" + repo.name + "+author:" + username + "+is:pr",
-      per_page: 100
+      per_page: 100,
+      page: pullRequestNum
     }).then(({data, headers, status}) => {
-      pullRequests = pullRequests.concat(data.items);
+      pullRequests = pullRequests.concat(data.items.map(pr => {
+        pr.repo = repo.name // add repo name as attribute of pull request
+        return pr
+      }));
       pullRequestNum = data.items.length;
     });
     
   } while (pullRequestNum >= 100);
 
   return pullRequests;
-}
\ No newline at end of file
+}
+
+// filter unnecessay attributes from pull requests and add status
+filterPullRequestAttributes = (pullRequests) => {
+  return pullRequests.map(pr => {
+    return {

```

Comment:
`return` keyword and function body can be omitted since the map callback function is a one-liner.
```
.....map(pr => {
      owner: pr.user.login,
      repo: pr.repo,
      number: pr.number,
      link: pr.html_url,
      title: pr.title,
      status: "pending"
});
```

Reply:
For some reason that creates a syntax error.